### PR TITLE
feat(matrix): implement e2ee readiness and owner invite autojoin

### PR DESCRIPTION
**Pull request by Athemis**
## Summary
- implement Matrix E2EE readiness handling with persistent crypto-store path setup, degraded-mode fallback, and structured readiness logging
- add owner-only invite autojoin (for DM and group invites) and wire owner Matrix aliases from gateway settings into `MatrixChannel`
- improve encrypted-event diagnostics, including explicit error logs when encrypted traffic is seen while E2EE is degraded
- reduce temporary high-volume debug noise in matrix/gateway/agent paths while keeping actionable warning/error logs
- finalize Matrix E2EE design + implementation plans with AC1-AC5 traceability and soft-gate follow-ups

## Test Plan
- [x] `uv run ruff check .`
- [x] `uv run ruff format . --check`
- [x] `uv run pytest -q`

## Summary by CodeRabbit

* **New Features**
  * End-to-end encryption for Matrix Direct Messages
  * Owner-restricted automatic room joins on invites

* **Documentation**
  * Added detailed design and implementation plans for Matrix E2EE and skills frontmatter delimiter hardening

* **Bug Fixes**
  * Hardened frontmatter delimiter parsing for skill YAML blocks

* **Improvements**
  * Better diagnostics, readiness reporting, and richer error logging across Matrix and gateway flows
* **Tests**
  * Added tests covering E2EE behavior, invite handling, crypto-store permissions, and frontmatter edge cases

### CODE REVIEW

File: squidbot/adapters/channels/matrix.py [hunk 1]
```diff
@@ -290,30 +377,88 @@ async def _handle_reaction(self, room: Any, event: Any) -> None:
...
+        room_id = getattr(room, "room_id", getattr(event, "room_id", ""))
+        join_resp = await self._client.join(room_id)
+        if isinstance(join_resp, nio.JoinError):
+            logger.error(
+                "MatrixChannel: auto-join failed room={} inviter={} err={}",
+                room_id,
+                sender,
+                join_resp,
+            )
+            return
```
[the remark this answers is not in the document]
Comment (Athemis) [reply squidbot/adapters/channels/matrix.py]: Fixed in commit 2c57648. Added guard for missing room_id and wrapped `await self._client.join(room_id)` in `try/except` with structured error logging (`room_id`, `inviter`, exception). Existing `nio.JoinError` handling remains in place, and `test_invite_join_exception_is_logged` was added to cover this path.
